internal/services/namedlocations/named_location_resource_test.go

```diff
@@ -0,0 +1,162 @@
+package namedlocations_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/manicminer/hamilton/msgraph"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azuread/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/utils"
+)
+
+type NamedLocationResource struct{}
+
+func TestAccNamedLocation_basicIP(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azuread_named_location", "test-ip")
+	r := NamedLocationResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basicIP(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccNamedLocation_completeIP(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azuread_named_location", "test-complete-ip")
+	r := NamedLocationResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.completeIP(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccNamedLocation_basicCountry(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azuread_named_location", "test-country")
+	r := NamedLocationResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basicCountry(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccNamedLocation_completeCountry(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azuread_named_location", "test-country")
```

Comment:
```suggestion
	data := acceptance.BuildTestData(t, "azuread_named_location", "test-complete-country")
```
Here's the bug. Change this and I get
```
> $ TF_LOG=TRACE TF_ACC=1 go test -v -timeout=2h ./internal/services/namedlocations                                                                             [±named-locations ●]
=== RUN   TestAccNamedLocation_basicIP
=== PAUSE TestAccNamedLocation_basicIP
=== RUN   TestAccNamedLocation_completeIP
=== PAUSE TestAccNamedLocation_completeIP
=== RUN   TestAccNamedLocation_basicCountry
=== PAUSE TestAccNamedLocation_basicCountry
=== RUN   TestAccNamedLocation_completeCountry
=== PAUSE TestAccNamedLocation_completeCountry
=== CONT  TestAccNamedLocation_basicIP
=== CONT  TestAccNamedLocation_completeCountry
=== CONT  TestAccNamedLocation_basicCountry
=== CONT  TestAccNamedLocation_completeIP
2021/06/02 19:23:29 [DEBUG] Importing Resource - parsing "0cc4a4a9-dfb3-425f-8ed1-70365cc7acdd"
2021/06/02 19:23:30 [DEBUG] Importing Resource - parsing "18b64bcf-75b4-40c0-865c-eecfb2016076"
2021/06/02 19:23:32 [DEBUG] Importing Resource - parsing "101afc99-0c08-48d7-b365-13fbf838d8bb"
2021/06/02 19:23:32 [DEBUG] Importing Resource - parsing "0dc61f25-134e-4bfe-853e-6db25f168dd6"
--- PASS: TestAccNamedLocation_basicIP (62.86s)
--- PASS: TestAccNamedLocation_basicCountry (62.96s)
--- PASS: TestAccNamedLocation_completeCountry (64.50s)
--- PASS: TestAccNamedLocation_completeIP (64.55s)
PASS
ok      github.com/hashicorp/terraform-provider-azuread/internal/services/namedlocations        64.563s
```
Put two update tests in there for basic > complete on both ip / country and should be good.

Reply:
So for the update test do all of the test data sets have to have the same resource label? Looking at other test functions for other resources I see that all of the resource labels are "test" but when I switch over to this I start getting inconsistent errors 
```
 TF_LOG=TRACE TF_ACC=1 go test -v -timeout=2h ./internal/services/namedlocations
=== RUN   TestAccNamedLocation_basicIP
=== PAUSE TestAccNamedLocation_basicIP
=== RUN   TestAccNamedLocation_completeIP
=== PAUSE TestAccNamedLocation_completeIP
=== RUN   TestAccNamedLocation_basicCountry
=== PAUSE TestAccNamedLocation_basicCountry
=== RUN   TestAccNamedLocation_completeCountry
=== PAUSE TestAccNamedLocation_completeCountry
=== RUN   TestAccNamedLocation_updateIP
=== PAUSE TestAccNamedLocation_updateIP
=== CONT  TestAccNamedLocation_basicIP
=== CONT  TestAccNamedLocation_completeCountry
=== CONT  TestAccNamedLocation_basicCountry
=== CONT  TestAccNamedLocation_completeIP
=== CONT  TestAccNamedLocation_updateIP
2021/06/02 19:42:12 [DEBUG] Named Location with Object ID "02f3ea6a-9dac-4b36-930e-75e93e8c361a" was not found - removing from state
2021/06/02 19:42:12 [WARN] Got error running Terraform: exit status 1

Error: Provider produced inconsistent result after apply

When applying changes to azuread_named_location.test, provider
"provider[\"registry.terraform.io/hashicorp/azuread\"]" produced an
unexpected new value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
=== CONT  TestAccNamedLocation_completeIP
    testcase.go:45: Step 1/2 error: Error running apply: exit status 1
        
        Error: Provider produced inconsistent result after apply
        
        When applying changes to azuread_named_location.test, provider
        "provider[\"registry.terraform.io/hashicorp/azuread\"]" produced an
        unexpected new value: Root resource was present, but now absent.
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
--- FAIL: TestAccNamedLocation_completeIP (4.16s)
2021/06/02 19:42:12 [DEBUG] Named Location with Object ID "00b71689-e358-4489-8b13-1a88eb2654ef" was not found - removing from state
2021/06/02 19:42:12 [WARN] Got error running Terraform: exit status 1

Error: Provider produced inconsistent result after apply

When applying changes to azuread_named_location.test, provider
"provider[\"registry.terraform.io/hashicorp/azuread\"]" produced an
unexpected new value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
=== CONT  TestAccNamedLocation_basicIP
    testcase.go:45: Step 1/2 error: Error running apply: exit status 1
        
        Error: Provider produced inconsistent result after apply
        
        When applying changes to azuread_named_location.test, provider
        "provider[\"registry.terraform.io/hashicorp/azuread\"]" produced an
        unexpected new value: Root resource was present, but now absent.
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
--- FAIL: TestAccNamedLocation_basicIP (4.58s)
=== CONT  TestAccNamedLocation_completeCountry
    testcase.go:45: Step 1/2 error: Check failed: Check 1/1 error: running exists func for "azuread_named_location.test": Named Location with object ID "1ed56c5c-fd26-4097-b409-268a44e51865" does not exist
=== CONT  TestAccNamedLocation_updateIP
    testcase.go:45: Step 1/6 error: Check failed: Check 1/1 error: running exists func for "azuread_named_location.test": Named Location with object ID "092314fc-10d1-4311-a1ff-04af26a696bb" does not exist
2021/06/02 19:42:32 [DEBUG] Named Location with ID "092314fc-10d1-4311-a1ff-04af26a696bb" already deleted
2021/06/02 19:42:33 [DEBUG] Importing Resource - parsing "1c163a0b-c3a9-404c-bcf7-c841a4e80bdc"
    testing_new.go:63: Error running post-test destroy, there may be dangling resources: "azuread_named_location.test" still exists
--- FAIL: TestAccNamedLocation_updateIP (44.98s)
--- FAIL: TestAccNamedLocation_completeCountry (45.37s)
--- PASS: TestAccNamedLocation_basicCountry (47.75s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-azuread/internal/services/namedlocations        47.761s
FAIL
```